components/scg/src/scg-struct.js

```diff
@@ -42,42 +42,48 @@ const SCgStructFromScTranslatorImpl = function (_editor, _sandbox) {
     };
 
     const doAppendBatch = function (batch) {
-
         for (let i in batch) {
             const task = batch[i];
             const addr = task[0];
             const type = task[1];
             let state = task[2];
             const level = task[3];
 
-            if (!state) state = SCgObjectState.FromMemory;
-
             delete addrsToAppendTasks[addr];
 
+            if (!state) state = SCgObjectState.FromMemory;
+
             let object = editor.scene.getObjectByScAddr(addr);
             if (object) {
-                if (!(sandbox.uniqueObjects && sandbox.uniqueObjects[addr])) {
+                if (!(sandbox.onceUpdatableObjects && sandbox.onceUpdatableObjects[addr])) {
                     object.setLevel(level);
                     object.setObjectState(state);
 
-                    if (sandbox.uniqueObjects) sandbox.uniqueObjects[addr] = object;
+                    if (sandbox.onceUpdatableObjects) sandbox.onceUpdatableObjects[addr] = object;
                 }
+
                 continue;
             }
 
             if (type & sc_type_node) {
                 object = SCg.Creator.createNode(type, randomPos(), '');
                 resolveIdtf(addr, object);
             } else if (type & sc_type_arc_mask) {
-                const bObj = editor.scene.getObjectByScAddr(task[4]);
-                const eObj = editor.scene.getObjectByScAddr(task[5]);
+                const sourceHash = task[4];
+                const targetHash = task[5];
+
+                // @TODO: implement reflexive sc.g-connectors viewing
+                if (sourceHash === targetHash) continue;
```

Comment:
```
struct = 
[*
	some_reflexive_set -> some_reflexive_set;;
	some_reflexive_node => nrel_itself: some_reflexive_node;;
*];;
```
renders incorrect when opening struct in scg mode

Reply:
Fixed
![image](https://github.com/ostis-ai/sc-web/assets/56268020/766cae59-6b80-44e4-af35-39266ca876c8)